rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -416,16 +416,34 @@ class DebruijnInterpreter[M[_], F[_]](
         _env.put(addr)
       }
 
-      def addUrn(newEnv: Env[Par], urn: String): Either[ReduceError, Env[Par]] =
-        urnMap.get(urn) match {
-          case Some(p) => newEnv.put(p).asRight[ReduceError]
-          case None    => ReduceError(s"Unknown urn for new: $urn").asLeft[Env[Par]]
-        }
+      def addUrn(
+          deployParameters: DeployParameters
+      )(newEnv: Env[Par], urn: String): Either[ReduceError, Env[Par]] =
+        if (urn == "rho:deployer:auth") {
+          val bs = deployParameters.userId.exprs.headOption match {
+            case Some(Expr(GByteArray(byteString))) => byteString
+            case Some(_) | None                     => ByteString.EMPTY
+          }
+          Right(
+            newEnv.put(
+              Par(
+                unforgeables = Vector(GDeployerAuth(bs))
+              )
+            )
+          )
```

Comment:
Please join those lines, and use `.asRight` consistently with surrounding code

Reply:
Done